app/src/main/java/com/zulip/android/activities/LoginActivity.java

```diff
@@ -265,6 +182,93 @@ public void onTaskFailure(String result) {
         }
     }
 
+    private void setupSignIN(String clientId) {
+        AuthorizationServiceConfiguration serviceConfiguration = new AuthorizationServiceConfiguration(
+                Uri.parse("https://accounts.google.com/o/oauth2/v2/auth") /* auth endpoint */,
+                Uri.parse("https://www.googleapis.com/oauth2/v4/token") /* token endpoint */
+        );
+
+        Uri redirectUri = Uri.parse("com.zulip.android:/oauth2callback");
```

Comment:
I'm trying this out with `http://zulip.tabbott.net` and getting an error about a redirect_uri_mismatch. Is there a canonical callback endpoint that the zulip servers already use?  If so, we should mimic that.